src/Controller/FightController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\User;
+use App\Entity\Attack;
+use App\Repository\UnicornRepository;
+use App\Repository\UserRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Attribute\CurrentUser;
+
+#[Route('/fight', name: 'app_fight_')]
+class FightController extends AbstractController
+{
+    #[Route('/', name: 'attack')]
+    public function attack(RequestStack $requestStack, UnicornRepository $unicornRepository): Response
+    {
+        $session = $requestStack->getSession();
+        $userUni = $session->get('userUnicorn');
+        $userUni = $unicornRepository->findOneBy(['id' => $userUni->getId()]);
+        $userAttacks = $userUni->getAttacks();
+        $session->set('round', 1);
+        $session->set('userScore', 100);
+        $session->set('opponentScore', 100);
+        return $this->render('fight/selectAttack.html.twig', ['attacks' => $userAttacks]);
+    }
+
+    #[Route('/round/', name: 'round')]
+    public function loopInRound(RequestStack $requestStack, UnicornRepository $unicornRepository): Response
+    {
+        $session = $requestStack->getSession();
+        $userUni = $session->get("userUnicorn");
+        $userUni = $unicornRepository->findOneBy(['id' => $userUni->getId()]);
+        $attacks = $userUni->getAttacks();
+        return $this->render('fight/selectAttack.html.twig', ['attacks' => $attacks]);
+    }
+
+    #[Route('/attackConfirm/{id}', name: 'attackConfirm')]
+    public function confirmAttack(
+        Attack $userAttack,
+        RequestStack $requestStack,
+        UnicornRepository $unicornRepository
+    ): Response {
+        $session = $requestStack->getSession();
+        $session->set('userAttack', $userAttack);
+        $oppoUni = $session->get('userUnicorn');
+        $oppoUni = $unicornRepository->findOneBy(['id' => $oppoUni->getId()]);
+        $attacks = $oppoUni->getAttacks();
+        $count = count($attacks);
+        $rnd = rand(0, $count - 1);
+        $attack = $attacks[$rnd];
+        $session->set("opponentAttack", $attack);
+        return $this->redirectToRoute('app_round');
+    }
+
+    #[Route('/{id}', name: 'indexUsers', methods: ['GET'])]
+    public function indexRandomUsers(int $id, #[CurrentUser] $user, UserRepository $userRepository): Response
```

Comment:
```suggestion
    public function indexRandomUsers(int $id, #[CurrentUser] ?User $user, UserRepository $userRepository): Response
```
That probably make Github Actions happy :)